File: algos/solutions/a1twopointers/JS/palindrome/helper_palindrome.js
// helper_palindrome.js

function showValidPalindromeDemo(isPalindrome) {
    console.log("========== Valid Palindrome Tests ==========");
    const testCases = [
        "RACEACAR",               // Mixed-case; after sanitization, should be palindrome if correct.
        "A",                      // Single character: always a palindrome.
        "ABCDEFGFEDCBA",          // Perfect palindrome.
        "ABC",                    // Not a palindrome.
        "ABCBA",                  // Palindrome (odd-length).
        "ABBA",                   // Palindrome (even-length).
        "No lemon, no melon",     // Palindrome with spaces.
        "Was it a car or a cat I saw?", // Palindrome with punctuation.
        "Able was I, I saw Elba", // Classic palindrome phrase.
        "12321",                  // Numeric palindrome.
        "123421",                 // Not a palindrome.
        "",                       // Empty string: considered a palindrome.
        "   "                     // Spaces only; when sanitized becomes empty → palindrome.
    ];

    testCases.forEach((s, index) => {
        console.log(`\nTest Case #${index + 1}`);
        console.log("-".repeat(80));
        console.log(`Input: '${s}' (length: ${s.length})`);
        console.log(`Is palindrome? → ${isPalindrome(s)}`);
        console.log("-".repeat(80));
    });
}

module.exports = { testValidPalindrome: showValidPalindromeDemo };

File: algos/solutions/a1twopointers/PY/palindrome/helper_palindrome.py
from algo import is_palindrome


def show_valid_palindrome_demo() -> None:
    print("========== Valid Palindrome Tests ==========")
    test_cases = [
        "RACEACAR",
        "A",
        "ABCDEFGFEDCBA",
        "ABC",
        "ABCBA",
        "ABBA",
        "No lemon, no melon",
        "Was it a car or a cat I saw?",
        "Able was I, I saw Elba",
        "12321",
        "123421",
        "",
        "   "
    ]
    for index, s in enumerate(test_cases, start=1):
        print(f"\nTest Case #{index}")
        print("-" * 80)
        print(f"Input: '{s}' (length: {len(s)})")
        print(f"Is palindrome? → {is_palindrome(s)}")
        print("-" * 80)


if __name__ == "__main__":
    show_valid_palindrome_demo()
